packages/visx-demo/src/components/GalleryTile.tsx

```diff
@@ -18,6 +18,43 @@ type Props<ExampleProps extends WidthAndHeight> = {
 const renderLinkWrapper = (url: string | undefined, node: React.ReactNode) =>
   url ? <Link href={url}>{node}</Link> : node;
 
+/**
+ * hook which returns if the ref was ever visible.
+ * used for better perf/not rendering all tiles on load.
+ */
+function useEverVisible() {
+  const ref = useRef();
+  const [everVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((entry) => {
+          setIsVisible((visible) => visible || entry.isIntersecting);
+        });
+      },
+      {
+        root: null, // viewport is the root
+        threshold: 0.01,
+      },
+    );
+
+    let curr;
+    if (ref.current) {
+      curr = ref.current;
+      observer.observe(ref.current);
+    }
+
+    return () => {
+      if (curr) {
+        observer.unobserve(curr);
+      }
+    };
+  }, []);
+
+  return { everVisible, ref };
+}
```

Comment:
What do you think about this hook instead?

```ts
function useIntersectionObserver() {
  const [entry, setEntry] = useState<IntersectionObserverEntry | null>(null);
  const prevObserver = useRef<IntersectionObserver | null>(null);

  const ref = (node: Element) => {
    if (prevObserver.current) {
      prevObserver.current.disconnect();
      prevObserver.current = null;
    }

    if (node.nodeType === Node.ELEMENT_NODE) {
      const observer = new IntersectionObserver(
        ([entry]) => {
          setEntry(entry);
        },
        { threshold: 0.01, root: null },
      );

      observer.observe(node);
      prevObserver.current = observer;
    }
  };

  return { entry, ref };
}
```


This could be a better approach for a few reasons:
1. I think `useIntersectionObserver` as a name feels like it more directly reflects its functionality, while `useEverVisible` is more abstract.
2. I think using `disconnect` is better because it completely removes the observer and stops watching all elements, ensuring proper cleanup, whereas `unobserve` only stops watching a specific element without cleaning up the observer itself, which can lead to memory leaks if not handled properly.
3. We can return the full `IntersectionObserverEntry`, giving access to more useful visibility data (maybe you don't want or care about this though).
4. The hook is easier to use with a single `customRef` function and fewer concerns about state tracking and effects.

[1]: https://github.com/uidotdev/usehooks/blob/90fbbb4cc085e74e50c36a62a5759a40c62bb98e/index.js#L512

Reply:
I dig it! this is a great suggestion for all the reasons you listed, I was just being lazy (hah!) :)